flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/mailbox/execution/MailboxProcessor.java

```diff
@@ -200,7 +200,7 @@ private boolean processMail(TaskMailbox mailbox) throws MailboxStateException, I
 			try {
 				maybeLetter.get().run();
 			} catch (Exception e) {
-				e.printStackTrace();
+				throw new IllegalStateException("Cannot process mail " + maybeLetter.get(), e);
```

Comment:
If we want to interrupt the execution for any exceptions during looping the letters, I am wondering whether this wrapped exception provides any helpful debug information. If not, we could remove `try..catch` clauses here for simple.

Reply:
Can we actually use the async exception handler? I stumbled upon that in another PR.

Reply:
From the description of `AsyncExceptionHandler`, it is proposed to used by other threads except the main task thread. AFAIK it is only used by `SplitReader` in `ContinuousFileReaderOperator` now. For this case if the `processMail` is executed in the main task thread, maybe it is out of scope of `AsyncExceptionHandler`.

Reply:
Actually I concerned two issues here:

- `IllegalStateException` seems not very accurate here.

- If we want to provide more message for debugging, maybe we could use `ExceptionUtils.rethrow(e, "Cannot process mail " + maybeLetter.get());`. But I am not sure whether `maybeLetter.get()` could give any helpful info here. If the debug info is useless, it seems simple to remove `try...catch` directly.

Reply:
`+1` to remove the `try..catch` here.

Reply:
I will remove commit and address it in #9772 .